autospec/tarball.py

```diff
@@ -499,10 +470,70 @@ def process_go_archives():
         if url_mod not in archives:
             go_archives.extend([url_mod, ':'])
         if url_zip not in archives:
-            go_archives.extend([url_zip, f'{ver}/'])
+            go_archives.extend([url_zip, ''])
         buildpattern.sources["godep"] += [url_info, url_mod, url_zip]
 
 
+def process_multiver_archives(main_src, multiver_archives):
+    """Set up multiversion archives."""
+    config_versions = config.parse_config_versions(build.download_path)
+    # Check if exist more than one version.
+    if len(config_versions) > 1:
+        for extraver in config_versions:
+            extraurl = config_versions[extraver]
+            if extraurl and extraurl != main_src.url:
+                buildpattern.sources["version"].append(extraurl)
+                multiver_archives.append(extraurl)
+                multiver_archives.append('')
+                set_multi_version(None)
+
+
+def process_archives(main_src, archives):
+    """Process extra sources needed by package.
+
+    This sources include: archives, go archives and multiversion.
+    """
+    go_archives = []
+    multiver_archives = []
+    src_objects = []
+
+    # Add extra archives and multiversion for Go packages
+    if os.path.basename(main_src.url) == "list":
+        process_go_archives(go_archives)
+    # Add multiversion for the rest of the patterns
+    else:
+        process_multiver_archives(main_src, multiver_archives)
+
+    full_archives = archives + go_archives + multiver_archives
+    # Download and extract full list
+    for arch_url, destination in zip(full_archives[::2], full_archives[1::2]):
+        src_path = check_or_get_file(arch_url, os.path.basename(arch_url), mode="a")
+        # Create source object and extract archive
+        archive = Source(arch_url, destination, src_path)
+        # Add archive prefix to list
+        buildpattern.archive_details[arch_url + "prefix"] = archive.prefix
```

Comment:
`archive.prefix` might be undefined here, should we test and possibly fall back to `archive.subdir`?

Reply:
So the code in specfiles tests if prefix isn't defined and uses essentially the subdir definition. Probably our change to cleanup that code litter in multiple places now, good catch.